### PR TITLE
export unicode symbol replace function - fixes #14

### LINKED ISSUE #14: support replacing unicode symbols in a string with fallbacks
See: https://github.com/sindresorhus/figures/issues/12#issuecomment-214500253

### PR DESCRIPTION
**Pull request by SamVerschueren**
Not sure if it can be done shorter. So feedback is more then welcome!

We should enable AppVeyor for these tests. I wanted to add the `.appveyor.yml` file but is it as simple as this? 

```yml
environment:
  matrix:
    - nodejs_version: '6'
    - nodejs_version: '4'
    - nodejs_version: '0.12'
    - nodejs_version: '0.10'
```

Or do we need a [test](https://github.com/sindresorhus/ava/blob/master/appveyor.yml#L20) script as well?

I compared this branch against the `tweaks` branch. Never done it before so I hope all goes well :). So better merge the `tweaks` first.

### CODE REVIEW

**Comment by SamVerschueren** [comment]
Added an appveyor.yml file but not sure if that's enough. I enabled AppVeyor for `figures` as well. Can you see that is is enabled @sindresorhus? Not very familiar with AppVeyor...

File: package.json [hunk 1]
```diff
@@ -34,5 +34,8 @@
   "devDependencies": {
     "ava": "*",
     "xo": "*"
+  },
+  "dependencies": {
```
Comment (kevva) [inline package.json]: Move it above the `devDependencies`.

**Comment by kevva** [comment]
Maybe add some example usage to the documentation?

File: package.json (hunk 1 above)
Comment (SamVerschueren) [inline package.json]: 👍

File: readme.md [hunk 2]
```diff
@@ -23,12 +23,29 @@ See the [source](index.js) for supported symbols.
...
 // On real OSes:  ✔︎
 // On Windows:    √
 ```
 
 
+## API
+
+### figures(input)
+
+Returns the `input` with replaced fallback unicode symbols.
```
Comment (SamVerschueren) [inline readme.md]: Not sure about this one. It seems like it's always replacing to fallback unicode symbols but it depends on the OS really. So one of the following might be better.

> Returns the `input` with replaced fallback unicode symbols on Windows.

> Returns the `input` with OS supported unicode symbols.

Or other suggestions are more then welcome as well.

**Comment by SamVerschueren** [comment]
@kevva Thanks for the feedback. Totally forgot to document it :).

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    - nodejs_version: '0.12'
+    - nodejs_version: '0.10'
+install:
+  - npm install
+shallow_clone: true
+clone_depth: 1
+test_script:
+  - node --version
+  - npm --version
+  - npm run test
```
Comment (sindresorhus) [inline appveyor.yml]: https://github.com/sindresorhus/appveyor-node/blob/master/appveyor.yml

Comment (SamVerschueren) [inline appveyor.yml]: Very nice! Thanks for that. WTF are all those lines even doing. Travis is so much more developer friendly.

File: readme.md [hunk 4]
```diff
@@ -23,12 +23,29 @@ See the [source](index.js) for supported symbols.
...
+
+#### input
+
+Type: `string`
+
+Value where the unicode symbols will be replaced with fallback symbols depending on the OS.
```
Comment (sindresorhus) [inline readme.md]: Value => String

File: readme.md (hunk 2 above)
Comment (sindresorhus) [inline readme.md]: > Returns the input with replaced fallback unicode symbols on Windows.

**Comment by SamVerschueren** [comment]
Had to add `escape-string-regexp` because of this error

> failed with "Invalid regular expression: /?⃝/: Nothing to repeat"

File: index.js [hunk 5]
```diff
@@ -124,4 +126,22 @@ if (platform === 'linux') {
...
+	if (figures === main) {
+		return str;
+	}
+
+	Object.keys(main).forEach(function (key) {
+		if (main[key] === figures[key]) {
+			return;
+		}
+
+		str = str.replace(new RegExp(escapeStringRegexp(main[key]), 'g'), figures[key]);
```
Comment (SamVerschueren) [inline index.js]: Anyone has an idea why I don't need the `m` flag here? Multiline replaces seems to work even without that flag.

**Comment by SamVerschueren** [comment]
If you provide me with the markdown for the AppVeyor badge, I can add it to the readme. Or you can do it afterwards.

File: index.js (hunk 5 above)
Comment (sindresorhus) [inline index.js]: No idea

Comment (sindresorhus) [inline index.js]: The `m` flag makes the regex work on a line-by-line basis if you use `^` and `$`, but we only replace one character globally, so it doesn't make a difference.

Comment (SamVerschueren) [inline index.js]: Aha, that makes sense. Thanks.

**Comment by sindresorhus** [comment]
Thanks Sam. I added the badge when merging ;)

**Comment by kevva** [comment]
Cool! :rocket:

**Comment by SamVerschueren** [comment]
Nice! Thanks guys for the feedback!

**Pull request closed without merging**